medusa/indexers/tvdbv2/tvdbv2_api.py

```diff
@@ -139,8 +141,12 @@ def _show_search(self, show, request_language='en'):
         """
         try:
             results = self.search_api.search_series_get(name=show, accept_language=request_language)
+        except ApiException as e:
+            raise IndexerShowNotFound(
+                'Show search failed in getting a result with reason: %s (%s)' % (e.reason, e.status)
+            )
         except Exception as e:
-            raise IndexerException('Show search failed in getting a result with error: %r', e)
+            raise IndexerException('Show search failed in getting a result with error: %r' % e)
```

Comment:
Just to clarify you are raising an exception that serves to do nothing but say that there was an exception and reraising a different exception.  This swallows important information (such as the traceback) and provides no additional information to the exception chain.  If it provided additional details to make the exception clearer, or if it raised a saner exception type it might be useful.  But raising a known exception type because you don't know what type of exception you might get is generally not a good practice.

Some examples:
from requests
```python
            try:
                connect, read = timeout
                timeout = TimeoutSauce(connect=connect, read=read)
            except ValueError as e:
                # this may raise a string formatting error.
                err = ("Invalid timeout {0}. Pass a (connect, read) "
                       "timeout tuple, or a single float to set "
                       "both timeouts to the same value".format(timeout))
                raise ValueError(err)
```
Here they are catching a ValueError, and yet reraising another ValueError because it provides context to the ValueError about what it actually means, and a traceback is unnecessary because there is nothing higher up the chain to debug and modify.

from our numdict.py:
```python
        try:
            return int(key)
        except (TypeError, ValueError):
            if key is not None:
                raise KeyError(key)
```
Since numdict acts like a dict, an invalid number raising a TypeError goes against expectations, thus we raise a KeyError instead.